subsys/lorawan/lorawan.c

```diff
@@ -0,0 +1,547 @@
+		return "Service unknown";
+	case LORAMAC_STATUS_PARAMETER_INVALID:
+		return "Parameter invalid";
+	case LORAMAC_STATUS_FREQUENCY_INVALID:
+		return "Frequency invalid";
+	case LORAMAC_STATUS_DATARATE_INVALID:
+		return "Datarate invalid";
+	case LORAMAC_STATUS_FREQ_AND_DR_INVALID:
+		return "Frequency or datarate invalid";
+	case LORAMAC_STATUS_NO_NETWORK_JOINED:
+		return "No network joined";
+	case LORAMAC_STATUS_LENGTH_ERROR:
+		return "Length error";
+	case LORAMAC_STATUS_REGION_NOT_SUPPORTED:
+		return "Region not supported";
+	case LORAMAC_STATUS_SKIPPED_APP_DATA:
+		return "Skipped APP data";
+	case LORAMAC_STATUS_DUTYCYCLE_RESTRICTED:
+		return "Duty-cycle restricted";
+	case LORAMAC_STATUS_NO_CHANNEL_FOUND:
+		return "No channel found";
+	case LORAMAC_STATUS_NO_FREE_CHANNEL_FOUND:
+		return "No free channel found";
+	case LORAMAC_STATUS_BUSY_BEACON_RESERVED_TIME:
+		return "Busy beacon reserved time";
+	case LORAMAC_STATUS_BUSY_PING_SLOT_WINDOW_TIME:
+		return "Busy ping-slot window time";
+	case LORAMAC_STATUS_BUSY_UPLINK_COLLISION:
+		return "Busy uplink collision";
+	case LORAMAC_STATUS_CRYPTO_ERROR:
+		return "Crypto error";
+	case LORAMAC_STATUS_FCNT_HANDLER_ERROR:
+		return "FCnt handler error";
+	case LORAMAC_STATUS_MAC_COMMAD_ERROR:
+		return "MAC command error";
+	case LORAMAC_STATUS_CLASS_B_ERROR:
+		return "ClassB error";
+	case LORAMAC_STATUS_CONFIRM_QUEUE_ERROR:
+		return "Confirm queue error";
+	case LORAMAC_STATUS_MC_GROUP_UNDEFINED:
+		return "Multicast group undefined";
+	case LORAMAC_STATUS_ERROR:
+		return "Unknown error";
+	default:
+		return NULL;
+	}
+}
+
+const char *eventinfo2str(int status)
+{
+	switch (status) {
+	case LORAMAC_EVENT_INFO_STATUS_OK:
+		return "OK";
+	case LORAMAC_EVENT_INFO_STATUS_ERROR:
+		return "Error";
+	case LORAMAC_EVENT_INFO_STATUS_TX_TIMEOUT:
+		return "Tx timeout";
+	case LORAMAC_EVENT_INFO_STATUS_RX1_TIMEOUT:
+		return "Rx 1 timeout";
+	case LORAMAC_EVENT_INFO_STATUS_RX2_TIMEOUT:
+		return "Rx 2 timeout";
+	case LORAMAC_EVENT_INFO_STATUS_RX1_ERROR:
+		return "Rx1 error";
+	case LORAMAC_EVENT_INFO_STATUS_RX2_ERROR:
+		return "Rx2 error";
+	case LORAMAC_EVENT_INFO_STATUS_JOIN_FAIL:
+		return "Join failed";
+	case LORAMAC_EVENT_INFO_STATUS_DOWNLINK_REPEATED:
+		return "Downlink repeated";
+	case LORAMAC_EVENT_INFO_STATUS_TX_DR_PAYLOAD_SIZE_ERROR:
+		return "Tx DR payload size error";
+	case LORAMAC_EVENT_INFO_STATUS_DOWNLINK_TOO_MANY_FRAMES_LOSS:
+		return "Downlink too many frames loss";
+	case LORAMAC_EVENT_INFO_STATUS_ADDRESS_FAIL:
+		return "Address fail";
+	case LORAMAC_EVENT_INFO_STATUS_MIC_FAIL:
+		return "MIC fail";
+	case LORAMAC_EVENT_INFO_STATUS_MULTICAST_FAIL:
+		return "Multicast fail";
+	case LORAMAC_EVENT_INFO_STATUS_BEACON_LOCKED:
+		return "Beacon locked";
+	case LORAMAC_EVENT_INFO_STATUS_BEACON_LOST:
+		return "Beacon lost";
+	case LORAMAC_EVENT_INFO_STATUS_BEACON_NOT_FOUND:
+		return "Beacon not found";
+	default:
+		return NULL;
```

Comment:
NULL can be passed to `log_strdup()` and then to `LOG_*()`. It will be safer to return something like `"Unknown"`. Same for `status2str()`.

Reply:
Fixed in successive commit!